x/ibc/core/02-client/keeper/grpc_query.go

```diff
@@ -157,19 +157,24 @@ func (q Keeper) ConsensusStates(c context.Context, req *types.QueryConsensusStat
 	consensusStates := []types.ConsensusStateWithHeight{}
 	store := prefix.NewStore(ctx.KVStore(q.storeKey), host.FullClientKey(req.ClientId, []byte(fmt.Sprintf("%s/", host.KeyConsensusStatePrefix))))
 
-	pageRes, err := query.Paginate(store, req.Pagination, func(key, value []byte) error {
+	pageRes, err := query.FilteredPaginate(store, req.Pagination, func(key, value []byte, accumulate bool) (bool, error) {
+		// filter any metadata stored under consensus state key
+		if strings.Contains(string(key), "/") {
+			return false, nil
```

Comment:
We could use bytes.Contains otherwise this []byte to string conversion is going to be very expensive.

Reply:
opened #8539